### PR TITLE
Encoder/PWM/GPIO

### PR DESCRIPTION
**Pull request by collinsc**
# Description
I added the requisite libraries to interface with our pwm breakout and jetson gpio. I wrote an interface to wrap them. 

The code looks big, but most of the code was from jetsonhacks for the gpio stuff. `PWMinterface` is the class we will be using to interface with our pwm controller, the `encoder.ino` is the arduino encoder sketch, and `quadrature.h` describes a generic encoder.
# Testing
This was a good candidate for a video demo:
https://youtu.be/Tewp5f_-oe8

### CODE REVIEW

**Review by RyanBerge — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -16,3 +16,8 @@ Installing the code requres, in addition to the standard `ros-kinetic-desktop-fu
  * `ros-kinetic-moveit`
  * `ros-kinetic-moveit-ros`
  * `ros-kinetic-cv-camera`
+ * `ros-kinetic-rosserial`
+ * `ros-kinetic-rosserial-arduino`
+
+# Arduino Libraries
+Some of our control software is installed on arduino. If that needs to be modified requires the Encoder library to be installed. https://www.pjrc.com/teensy/td_libs_Encoder.html
```
Comment (Jebbs) [inline README.md]: Maybe change "If that needs to be modified requires the Encoder library to be installed." to "Modifications require the Encoder library to be installed." Or something like that.

File: src/tfr_control/arduino/encoder/encoder.ino [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+//pin constants
+const int GEARBOX_LEFT_A = 2;
+const int GEARBOX_LEFT_B = 3;
+
+//encoders
+Quadrature gearbox_left(GEARBOX_CPR, GEARBOX_LEFT_A, GEARBOX_LEFT_B);
+
+//algorithm constants
+float rate(1/50.0);
```
Comment (Jebbs) [inline src/tfr_control/arduino/encoder/encoder.ino]: should this also be marked as `const`?

File: src/tfr_control/src/pwm_bridge.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
+/**
+ * Simple example to bring up a node and blink an led at address 0 on the i2c
+ * pwm bus and demo the driver.
```
Comment (Jebbs) [inline src/tfr_control/src/pwm_bridge.cpp]: Was leaving this here intentional? It doesn't look like this file does any blinking.

**Review by Jebbs — COMMENTED** [review]
I didn't look over the stuff that came from JetsonHacks, which I figured should be fine.

This is mostly a proof of concept, right? I ask because `pwm_bridge.cpp` only references the left tread and the blink example is still included.

File: README.md (hunk 1 above)
Comment (collinsc) [reply README.md]: thats fine

File: src/tfr_control/arduino/encoder/encoder.ino (hunk 2 above)
Comment (collinsc) [reply src/tfr_control/arduino/encoder/encoder.ino]: it can

File: src/tfr_control/src/pwm_bridge.cpp (hunk 3 above)
Comment (collinsc) [reply src/tfr_control/src/pwm_bridge.cpp]: nope

**Comment by collinsc** [comment]
@Jebbs changes implemented, and yes this will be expanded upon on future iterations.

**Review by Jebbs — APPROVED** [review]
Looks good, thanks!

**Pull request merged**